staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -146,6 +146,27 @@ func encodeManagedFields(managedFields fieldpath.ManagedFields) (encodedManagedF
 		}
 		encodedManagedFields = append(encodedManagedFields, *v)
 	}
+	return sortEncodedManagedFields(encodedManagedFields)
+}
+
+func sortEncodedManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (sortedManagedFields []metav1.ManagedFieldsEntry, err error) {
+	sort.Slice(encodedManagedFields, func(i, j int) bool {
+		p, q := encodedManagedFields[i], encodedManagedFields[j]
+
+		if p.Operation != q.Operation {
+			return p.Operation < q.Operation
+		}
+
+		if p.Time == nil || q.Time == nil {
+			return false
```

Comment:
Should Manager field be compared in this case ?

Reply:
that's a good point, actually they should be compared if they are both nil.

Reply:
Or rather:
we're only going to see that if operations are the same, which means that they are both update or  both apply. If they are both apply, we shouldn't expect any to be nil, and we want to compare by manager. If they are update then we should expect neither to be nil, and compare first by time and then by manager.

Reply:
I thought I saw a test testing that, but I must have been wrong ... We need to have a test for that.

Reply:
Maybe change the check of Time against nil to assertion since you don't expect it to be nil.

Reply:
We already got an assertion for nil managers here: https://github.com/kwiesmueller/kubernetes/blob/4394bf779800710e67beae9bddde4bb5425ce039/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go#L249 before this gets called. But yes could put one here as well.

And we do expect time to be nil on Apply operations.